recipes/xerces-c/all/conanfile.py

```diff
@@ -99,16 +98,18 @@ def validate(self):
         self._validate("mutex_manager", "windows", ("Windows", ))
 
     def build_requirements(self):
-        if hasattr(self, "settings_build") and self.options.message_loader == "icu":
-            self.tool_requires("icu/71.1")
+        if self.options.message_loader == "icu" and not can_run(self):
```

Comment:
To avoid extra builds in case of native build (options and icu version in build & host context may be different, leading to unecessary build of another package of icu in build context). Moreover build requirements are more fragile than requirements because they are exposed to versions conflicts without any way to solve conflict.

Reply:
This actually makes a lot of sense... I've adde to out weekly meeting to review with the team but I can get behind this 👍